### DISCUSSION TITLE
How to deploy angular project with prisma

### DISCUSSION THREAD

**Discussion by jschank** [prisma/prisma#14163] [Q&A]
I have project, built in Angular 14, and NestJS. I deploy this to a VM on Amazon AWS. And I am trying to understand how to deploy the prisma client that gets generated to access my Sqlite3 database.

As I understand it, prisma is installed to my node_modules. And when I run something like `npx prisma migrate`, in my project, on my development machine, then prisma correctly updates its client and my interactions with the data, will succeed. 

However, when I deploy, I seem to need to have my node_modules available, so that prisma can do its CLI things on the deployment machine. If the node_modules is not available, then my deployed app cannot interact with the database.

So my question is: What is the correct workflow for deploying my application, and how does my deployed application "find" the generated prisma client. In order to communicate with the production DB?

Am I expected to copy the prisma client into my dist folder (for instance) ?
Do I need to pull down a copy of the prod db, run prisma migrations locally, and then push the db back up to the deployment machine? (this is simple considering that Sqlite3 is a single file, but for something like MySql this would seem untenable.

Thanks for any suggestions
jschank

**Comment by austincrim** [comment]
Hey @jschank 👋,

What does your deployment setup look like now?

It is generally true that there is a step in your CI pipeline that runs `npm install` (or equivalent), after that you should have the `prisma` CLI available to use for migration commands.

**Reply by jschank** [reply]
I don't currently have a CI pipeline going. I manually tar up my Dist folder, and then untar it on the destination server.
I typically also upload my package.json, and package-lock.json, and would run $ npm i, on the server to update the node_modules, there. But it seems like I should not need to do that. I thought (but could be wrong) that all the dependencies that are required are compiled down into the dist folder.

It sounds like the expected process, is that I should clone my repo onto the deployment machine, and then build there. And just hook things like the web server and node, into the correct parts of the dist folder on the deployment machine. But again, that seems like overkill.

**Reply by jschank** [reply]
Also... If I rename the node_modules folder on the deployment machine, to somthing else (so it isn't found) and restart my servers. (one for nest js, and one for the application) Then, things seem to still work, EXCEPT that prisma doesn't seem able to communicate with the database. I theorize, that this is because the prisma client, which is generated, it kept in the node_modules folder. 

So I guess, one question, is: Is there a way to have the prismaClient generated into the dist folder. (or otherwise included in the build)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
